### PR TITLE
pid1: send shutdown type and reboot argument to supervisor via sd_notify()

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

File: src/systemctl/systemctl-start-special.c [hunk 1]
```diff
@@ -155,14 +155,16 @@ int verb_start_special(int argc, char *argv[], void *userdata) {
...
 
-        if (a == ACTION_REBOOT) {
-                if (arg_reboot_argument) {
-                        r = update_reboot_parameter_and_warn(arg_reboot_argument, false);
-                        if (r < 0)
-                                return r;
-                }
+        if (arg_reboot_argument && IN_SET(a, ACTION_HALT, ACTION_POWEROFF, ACTION_REBOOT, ACTION_KEXEC)) {
+                /* If we are going through an action that involves systemd-shutdown, let's set the reboot
+                 * parameter, even if it's not a regulare reboot. After all we nowadays send the string to
```
Comment (YHNdnzj) [inline src/systemctl/systemctl-start-special.c]:
```suggestion
                 * parameter, even if it's not a regular reboot. After all we nowadays send the string to
```

File: src/shutdown/shutdown.c [hunk 2]
```diff
@@ -333,6 +333,24 @@ static void init_watchdog(void) {
...
+        int r;
+
+        r = read_reboot_parameter(&reboot_parameter);
+        if (r < 0 && r != -ENOENT)
+                log_debug_errno(r, "Failed to read reboot parameter, ignoring: %m");
+
+        if (reboot_parameter)
+                (void) sd_notifyf(/* unset_environment= */ false,
+                                  "X_SYSTEMD_SHUTDOWN=%s\n"
+                                  "X_SYSTEMD_REBOOT_PARAMETER=%s", arg_verb, reboot_parameter);
```
Comment (YHNdnzj) [inline src/shutdown/shutdown.c]: Break line before arg_verb?

**Review by YHNdnzj — APPROVED** [review]

**Review by keszybz — APPROVED** [review]
LGTM.

**Comment by keszybz** [comment]
centos ci (arch linux):
```
11:48:43 [  737.048336] kernel: INFO: task 3:4075 blocked for more than 614 seconds.
11:48:43 [  737.048338] kernel:       Not tainted 6.8.7-arch1-1 #1
11:48:43 [  737.048339] kernel: "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
```
… and TEST-02-UNITTESTS failed. Doesn't seem related.

testing-farm:fedora-rawhide-x86_64 is listed as failed, but copr thinks the build succeeded.

**Pull request merged**